### ISSUE TITLE
Investigating OS Error during HTML file write

### ISSUE THREAD

**Issue by Mulan-94** [ratt-ru/ragavi#78]
This investigation led to  [this bokeh issue:](https://github.com/bokeh/bokeh/pull/7539/files/bed658ce8efd4793ded7d7ccf6b39513a81ffbab#diff-22941b79bd893a19141807296c25c0ce)

and to the [package tempfile:](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)

This [stack abuse post](https://stackabuse.com/the-python-tempfile-module/#:~:text=The%20tempfile.,create%20a%20file%20in%20it) gives some further information.
Tempfile is a default python package that scans the following list of directories :

- The directory named by the TMPDIR environment variable.
- The directory named by the TEMP environment variable.
- The directory named by the TMP environment variable
- Platform-specific directories:
- On windows, C:\TEMP, C:\TMP, \TEMP, and \TMP, in the same order.
  - On other platforms, /tmp, /var/tmp, and /usr/tmp, in the same order.
  - The current working directory.

To get the default temp files location: `tempfile.gettempdir()`
To set tempdir: `tempfile.tempdir = "/bla/bla"`

I'm going to change the temp dir location to see if this helps

**Comment by o-smirnov** [comment]
Yep, I think you've found the culprit! Cross-linking to https://github.com/caracal-pipeline/caracal/issues/1200

**Comment by KshitijT** [comment]
Yay ! 
So would it be a separate directory than the one set by , e.g. 
`export SINGULARITY_TMPDIR=/idia/projects/fornax/.singularity_tmp` ?

**Comment by o-smirnov** [comment]
Yeah, ``SINGULARITY_TMPDIR`` would be respected by singularity itself, but not necessarily by the container inhabitants. I've added an issue on Stimela which would be a general defensive solution. In the meantime, ragavi should contain its own fix of course.
